OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -124,6 +126,13 @@ public void Killed(Actor self, AttackInfo e)
 				attackerStats.KillsCost += cost;
 				defenderStats.DeathsCost += cost;
 			}
+
+			var experienceInfo = self.Info.TraitInfo<GivesExperienceInfo>();
+			if (experienceInfo != null && experienceInfo.GivesPlayerExperience)
+		    {
+				var exp = self.Info.TraitInfo<ValuedInfo>().Cost;
```

Comment:
But that line is under a HasTraitInfo check. 

I'll take Paul's comment about experience into account though, so it will be a newly declared variable anyways.